### ISSUE TITLE
If letterSpacing and scale are used together,  position is different from the web browser canvas

### ISSUE THREAD

**Issue by loveloki** [samizdatco/skia-canvas#210]
## Description

I want drawText with letterSpacing and scale.

So I need calculate new position and drawText.

The core code is this:
```js
function drawTextWithScale(text, x, y, scaleX, scaleY, letterSpacing) {
  ctx.save();
  ctx.setTransform(scaleX, 0, 0, scaleY, 0, 0);
  ctx.letterSpacing = letterSpacing;

  ctx.textBaseline = "top";

  const newX = x / scaleX;
  const newY = y / scaleY;

  ctx.fillText(text, newX, newY);

  ctx.resetTransform();
  ctx.restore();
}
```

Code see full code in [scale.js](https://github.com/loveloki/canvas-test/blob/main/src/stage/scale.js)

And the final rendered position is different from the web browser canvas.

## Preview

[preview different canvas render result](https://canvas-test-nine.vercel.app/)

## Sample screenshot:

![image](https://github.com/user-attachments/assets/bcdcbb7b-8b7f-4fa6-b23f-f42b7be64899)

## Summarize

1. `node-canvas` does not support letterSpacing with scale, but the position is same with the browser web canvas(maybe that's why it's in the right position).
2. `skia-canvas` support letterSpacing with scale, but **the first letter position** is different from the web browser canvas.

## Question

1. I'm not very familiar with the math behind it, how should I calculate position to match the web canvas result?
2. Is this caused by different engine calculations logic? Whether (how) the results will be modified for web canvas? If you can help me, I'll be happy to solve it

**Comment by samizdatco** [comment]
I think the differences you're seeing come down to whether each canvas implementation supports the `letterSpacing` attribute or not. At the moment I believe Chrome is the only browser that does, so make sure you're using that as your comparison point. 

Here are two test strings drawn in Chrome with `ctx.letterSpacing = '30px'`; the first has no transform and the second is drawn with `ctx.scale(2,1)`:
<img width="755" alt="image" src="https://github.com/user-attachments/assets/7e1b5d1e-c549-4967-8afe-02b7255ee98c" />

I added the magenta rectangles in Illustrator, but the top ones are 30px wide and the bottom are 60px wide (due to the scale transform). So (to my eye at least) it's working as expected?

**Comment by loveloki** [comment]
Yes I use Chrome to comparison result. [preview different canvas render result](https://canvas-test-nine.vercel.app/)


The calculated interval is expected right to me.

I studied it again after raising the issue, then I think **the position of the first letter position** is wrong. In Chrome, the position of the first word will not be changed.(Accordingly I just updated the description.)

View the example:

![image](https://github.com/user-attachments/assets/bcdcbb7b-8b7f-4fa6-b23f-f42b7be64899)


Thanks to your reminder, I tested it in Indesign:

![image](https://github.com/user-attachments/assets/5c629825-15cd-4ab3-a057-3dd3e60dab9c)
![image](https://github.com/user-attachments/assets/f5e705c2-b9dc-4ca0-bcc7-5d1ca16bc585)

As you can see, the first letter does not change its position and only affects subsequent letters.

It follows that the position of the initial letter should not be changed by letterspcing.

**Comment by samizdatco** [comment]
Version 2.0.2 includes a fix to remove the unexpected ½ letterspace Skia adds to the beginning of each line.
